src/solver/optimisation/constraints/Group1.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include "ConstraintGroup.h"
+#include "AreaBalance.h"
+#include "FictitiousLoad.h"
+#include "ShortTermStorageLevel.h"
+#include "FlowDissociation.h"
+#include "BindingConstraintHour.h"
+
+class Group1 : public ConstraintGroup
+{
+public:
+    using ConstraintGroup::ConstraintGroup;
+
+        void Build() override;
+
+private:
+    AreaBalanceData GetAreaBalanceData();
```

Comment:
As a consequence of a previous comment of the current file, **AreaBalanceData** seems useless, as its content should be stored a constraint class (**AreaBalance**).
Same for other data classes here.

Reply:
et tout ce monde dans la methode add?
` std::vector<CORRESPONDANCES_DES_CONTRAINTES>& CorrespondanceCntNativesCntOptim;
    const std::vector<int>& IndexDebutIntercoOrigine;
    const std::vector<int>& IndexSuivantIntercoOrigine;
    const std::vector<int>& IndexDebutIntercoExtremite;
    const std::vector<int>& IndexSuivantIntercoExtremite;
    const std::vector<PALIERS_THERMIQUES>& PaliersThermiquesDuPays;
    const std::vector<::ShortTermStorage::AREA_INPUT>& ShortTermStorage;`